pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -636,7 +636,7 @@ func (m *kubeGenericRuntimeManager) killContainer(pod *v1.Pod, containerID kubec
 	if gracePeriod < minimumGracePeriodInSeconds {
 		gracePeriod = minimumGracePeriodInSeconds
 	}
-	if gracePeriodOverride != nil {
+	if gracePeriodOverride != nil && *gracePeriodOverride >= 0 {
 		gracePeriod = *gracePeriodOverride
 		klog.V(3).Infof("Killing container %q, but using a %d second grace period override", containerID, gracePeriod)
 	}
```

Comment:
This is a nice addition. 

In practice, looks like `gracePeriodOverride` won't be set to negative value.

https://github.com/kubernetes/kubernetes/blob/acb43c7c4ac565eb2b51f98dcce32b5879383a19/pkg/kubelet/eviction/eviction_manager.go#L381-L384

Reply:
That's very helpful. Thank.